config/proto_conversions.go

```diff
@@ -209,21 +209,25 @@ func ResourceLevelServiceConfigFromProto(proto *pb.ResourceLevelServiceConfig) (
 }
 
 // FrameConfigToProto converts Frame to proto equivalent.
-func FrameConfigToProto(frame Frame) (*pb.Frame, error) {
+func FrameConfigToProto(frame referenceframe.LinkConfig) (*pb.Frame, error) {
+	pose, err := frame.Pose()
+	if err != nil {
+		return nil, err
+	}
 	proto := pb.Frame{
 		Parent: frame.Parent,
 		Translation: &pb.Translation{
-			X: frame.Translation.X,
-			Y: frame.Translation.Y,
-			Z: frame.Translation.Z,
+			X: pose.Point().X,
+			Y: pose.Point().Y,
+			Z: pose.Point().Z,
 		},
 	}
 
 	var orientation pb.Orientation
 
-	switch oType := frame.Orientation.(type) {
+	switch oType := pose.Orientation().(type) {
```

Comment:
Nit: Consider assigning pose.Orientation() and checking type and doing casting on the assigned variable?